bus-mapping/src/circuit_input_builder.rs

```diff
@@ -902,6 +923,38 @@ impl<'a> CircuitInputStateRef<'a> {
         Ok(())
     }
 
+    /// Push a [`MemoryOp`] into the [`OperationContainer`] with the next
+    /// [`RWCounter`] and `call_id`, and then adds a reference to
+    /// the stored operation ([`OperationRef`]) inside the bus-mapping
+    /// instance of the current [`ExecStep`].  Then increase the `block_ctx`
+    /// [`RWCounter`] by one.
+    pub fn memory_read(
+        &mut self,
+        step: &mut ExecStep,
+        address: MemoryAddress,
+        value: u8,
+    ) -> Result<(), Error> {
+        let call_id = self.call()?.call_id;
+        self.push_op(step, RW::READ, MemoryOp::new(call_id, address, value));
+        Ok(())
+    }
+
+    /// Push a [`MemoryOp`] into the [`OperationContainer`] with the next
```

Comment:
```suggestion
    /// Push a write type [`MemoryOp`] into the [`OperationContainer`] with the next
```

Reply:
updated in https://github.com/appliedzkp/zkevm-circuits/pull/497/commits/206d5fd8df436421e37e2a80bfa8b2965c487361 thanks